### PR TITLE
Add parameter validation for the ECDH module

### PR DESCRIPTION
**Pull request by hanno-arm**
__Summary:__ Based on #1643, this PR adds parameter validation for the ECDH module.

### CODE REVIEW

**Comment by hanno-arm** [comment]
Fixed `doxygen.sh` failure inn 0990ca7.

File: library/ecdh.c [hunk 1]
```diff
@@ -57,6 +64,10 @@ static int ecdh_gen_public_restartable( mbedtls_ecp_group *grp,
                     mbedtls_ecp_restart_ctx *rs_ctx )
 {
     int ret;
+    ECDH_VALIDATE_RET( grp != NULL );
```
Comment (gilles-peskine-arm) [inline library/ecdh.c]: Not a blocker: this is an internal function, the null checks are redundant.

File: library/ecdh.c [hunk 2]
```diff
@@ -488,9 +510,9 @@ int mbedtls_ecdh_make_public( mbedtls_ecdh_context *ctx, size_t *olen,
                               void *p_rng )
 {
     int restart_enabled = 0;
-
-    if( ctx == NULL )
-        return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
+    ECDH_VALIDATE_RET( ctx != NULL );
+    ECDH_VALIDATE_RET( olen != NULL );
+    ECDH_VALIDATE_RET( buf != NULL );
```
Comment (gilles-peskine-arm) [inline library/ecdh.c]: Missing: `f_rng != NULL`

File: include/mbedtls/ecdh.h [hunk 3]
```diff
@@ -381,7 +422,7 @@ int mbedtls_ecdh_calc_secret( mbedtls_ecdh_context *ctx, size_t *olen,
  *                  computations once enabled, except by free-ing the context,
  *                  which cancels possible in-progress operations.
  *
- * \param ctx       The ECDH context.
+ * \param ctx       The ECDH context to use. This must be initialized.
```
[the remark this answers is not in the document]
Comment (hanno-arm) [reply include/mbedtls/ecdh.h]: This is a deliberate abbreviation for "points to an initialized context", hence is stronger than not being NULL. This wording is used in the other PRs, too.

File: library/ecdh.c (hunk 1 above)
Comment (hanno-arm) [reply library/ecdh.c]: Fixing.

File: library/ecdh.c (hunk 2 above)
Comment (hanno-arm) [reply library/ecdh.c]: Fixing.

**Comment by hanno-arm** [comment]
@gilles-peskine-arm I have addressed your feedback. Please re-review.

**Comment by sbutcher-arm** [comment]
Reviewed, approved, and passes CI. Labelling as 'ready to merge'.

**Pull request merged**